Implemented Emojis, Reactions, and bug fixes

ZOrangeBandit opened the pull request.

ZOrangeBandit committed:
Implemented Emojis, client-sided Reactions, bugfixes

ZOrangeBandit committed:
Fully implemented Reactions

ZOrangeBandit committed:
Changed emoji pickers for reactions

turtle11235 commented:
It looks like when I change my screen name it doesn't update the database so it thinks all my previous messages are from someone else

Commit:
Added Chat Toggle switch (not fully implemented)
Removed unneccisary EmojiPickerButton
Fixed message and emoji names not updateding when changing displayName

Commit:
Made reactions one lined, made style match on each side

Commit:
Changed message box to a textarea, implemented enter submission and multiline messages

Commit:
Satisified React's warnings

turtle11235 commented:
looks good to me
